src/modify.jl

```diff
@@ -1,7 +1,10 @@
 ###### update ####################
 
 function update{T,N,D}(ta::TimeArray{T,N,D}, tstamp::D, val::Vector{T})
-    if tstamp < maximum(ta.timestamp)
+    if length(ta) == 0
+        t    = vcat(tstamp)
```

Comment:
There are constructors already in place to take care of a single `tstamp`. So you really just want to construct the TimeArray with a simple `TimeArray(tstamp, val)` and you're done. 

I'm not sure if you noticed by I assigned this issue to myself, just haven't gotten to it yet. And it will require a few more tests as well.